### DISCUSSION TITLE
Example using SWC with path aliases?

### DISCUSSION THREAD

**Discussion by jasonkuhrt** [TypeStrong/ts-node#1640] [Q&A] [answered]
I haven't been able to get ts-node + SWC + TS path aliases to work. My simple PR/attempt is here https://github.com/jasonkuhrt/template-typescript-lib/pull/55. Anyone can point out the issue or a different working example? 

Thanks!

**Comment by cspotcode** [comment, marked as answer]
For starters, how do you use path aliases without swc?  We can build off of that.

**Reply by jasonkuhrt** [reply]
@cspotcode You can see how it is currently done in the main branch.

1. Install dep `typescript-transform-paths` 
https://github.com/jasonkuhrt/template-typescript-lib/blob/main/package.json#L66
2. Config ts-node to use its register hook `"require": ["typescript-transform-paths/register"],` 
https://github.com/jasonkuhrt/template-typescript-lib/blob/main/tsconfig.json#L4

**Reply by cspotcode** [reply]
Ok that's what I suspected, it's a custom TS transformer.

Check out https://typestrong.org/ts-node/docs/paths, #1585, and #1450

**Reply by jasonkuhrt** [reply]
@cspotcode I'm not sure I follow. I already have it working without SWC. I assumed the issue would be on the SWC level.

**Reply by cspotcode** [reply]
I assume typescript-transform-paths works by hooking into the typescript compiler, but if you replace that compiler with swc, then it can't do that transformation.  The links I shared relate to solutions for path mapping that do not require hooking into the TS compiler, so they're compatible with swc.

**Reply by jasonkuhrt** [reply]
@cspotcode Yeah I understood that the previous solution was using TSC which SWC does not. However based on the links in my PR I expected SWC to support path aliasing out of the box. For example: https://github.com/swc-project/swc/issues/1935 and https://swc.rs/docs/configuration/compilation#jscpaths.

But now I see your point about your own links that those can be used in a way that is not reliant on TSC. I'll give that a shot!

That said, I feel like this should be possible with vanilla SWC... 🤔

**Reply by jasonkuhrt** [reply]
Thanks @cspotcode https://typestrong.org/ts-node/docs/paths/ worked for me for now 👏

**Reply by cspotcode** [reply]
Yeah, the problem with handling mappings in the TS->JS transformation layer, is that we've got a bunch of QOL resolver behaviors planned: #1514  I don't think it'd be feasible to implement all of them in a way that meets the requirements:

a) done within the transformer
b) compatible with all transformers: typescript, swc, other future third-party transformers
c) compatible with pre-compiled code (if a project has a mix of compiled ts and non-compiled js, we probably want these resolver behaviors to apply to both)

Instead we opt to implement resolutions within the node engine's own resolver.
